src/s_bsd.c

```diff
@@ -1363,18 +1362,6 @@ aClient *add_connection(aListener *lptr, int fd)
     acptr->lstn = lptr;
     add_client_to_list(acptr);
 
-    ban = check_userbanned(acptr, UBAN_IP|UBAN_CIDR4|UBAN_WILDUSER, 0);
-    if(ban)
-    {
-        int loc = (ban->flags & UBAN_LOCAL) ? 1 : 0;
-        
-        ircstp->is_ref++;
-        ircstp->is_ref_1++;
-        exit_banned_client(acptr, loc, loc ? 'K' : 'A', ban->reason, 0);
-
-        return NULL;
-    }
-
```

Comment:
This looks like check_userbanned will be postponed until SERVER or USER is parsed even for non-SSL connections. Couldn't we just silently return here if it is a banned SSL connection, and add another ban check when the SSL handshake is finished?

Reply:
True. By skipping this part, one less look through the banlist for a match, especially if the ban is placed on user@host, as we don't have the user info here yet.